hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/cache/CacheManager.java

```diff
@@ -0,0 +1,312 @@
+    private static final String RESET_PUBLISH_CACHE_ENV = "RESET_PUBLISH_CACHE";
+
+    private static final Set<String> DATA_ID_PREFIXES = new HashSet<>();
+
+    static {
+        // TODO: For now just use the built-in H Metrics prefixes as a OOB convenience, but in the
+        //       future this should maybe be configurable where a list of prefixes could be supplied
+        //       via system prop.
+        DATA_ID_PREFIXES.add("hm_a_");  // Metrics AVAILABILITY
+        DATA_ID_PREFIXES.add("hm_c_");  // Metrics COUNTER
+        DATA_ID_PREFIXES.add("hm_cr_"); // Metrics COUNTER_RATE
+        DATA_ID_PREFIXES.add("hm_g_");  // Metrics GAUGE
+        DATA_ID_PREFIXES.add("hm_gr_"); // Metrics GAUGE_RATE
+        DATA_ID_PREFIXES.add("hm_s_");  // Metrics STRING
+        DATA_ID_PREFIXES.add("hm_u_");  // Metrics UNDEFINED
+    }
+
+    private static boolean PUBLISH_REQUIRE_PREFIX;
+
+    @EJB
+    PropertiesService properties;
+
+    @EJB
+    DefinitionsService definitions;
+
+    @Resource(lookup = "java:jboss/infinispan/cache/hawkular-alerts/publish")
+    private Cache<CacheKey, String> publishCache;
+
+    @PostConstruct
+    public void init() {
+        PUBLISH_REQUIRE_PREFIX = Boolean.parseBoolean(properties.getProperty(PUBLISH_REQUIRE_PREFIX_PROP,
+                PUBLISH_REQUIRE_PREFIX_ENV, "true"));
+        boolean disablePublish = Boolean.parseBoolean(properties.getProperty(DISABLE_PUBLISH_FILTERING_PROP,
+                DISABLE_PUBLISH_FILTERING_ENV, "false"));
+        boolean resetCache = Boolean.parseBoolean(properties.getProperty(RESET_PUBLISH_CACHE_PROP,
+                RESET_PUBLISH_CACHE_ENV, "true"));
+        if (!disablePublish) {
+            if (resetCache) {
+                msgLog.warnClearPublishCache();
+                publishCache.clear();
+            }
+            msgLog.infoInitPublishCache();
+            updateActiveIds();
+            definitions.registerListener(e -> {
+                updateActiveIds();
+            }, CONDITION_CHANGE, TRIGGER_REMOVE);
+        } else {
+            msgLog.warnDisabledPublishCache();
+        }
+    }
+
+    // TODO REMOVE THESE WHEN THE JMS LISTENERS GO AWAY !!!
+    public Set<DataIdKey> getActiveDataIds() {
+        return Collections.emptySet();
+    }
+
+    public Set<DataIdKey> getActiveAvailabilityIds() {
+        return Collections.emptySet();
+    }
+
+    private synchronized void updateActiveIds() {
+        try {
+            CacheSet<CacheKey> currentlyPublished = publishCache.keySet();
+
+            log.debugf("Published before update=%s", currentlyPublished.size());
+            if (log.isTraceEnabled()) {
+                publishCache.entrySet().stream().forEach(e -> log.tracef("Published: %s", e.getValue()));
+            }
+
+            // This will include group trigger conditions, which don't need to be published but it's
+            // not easy to efficiently filter them out.
```

Comment:
I was thinking about this more yesterday when looking at the addpendingData filtering and that logic, and actually I think we *do* want to include the group trigger data ids, and I think I want to update or remove this comment in the PR code.  The reason is data-driven group triggers.  I actually had forgotten about this sort-of-cool feature.  But if you recall (or go back and read the docs like I had to do :)  We can generate member triggers based on the "source" values provided  on incoming Data.  In this use case the dataIds *are the generic group trigger dataids*!  So we actually do want to process Data with the group trigger dataIds!